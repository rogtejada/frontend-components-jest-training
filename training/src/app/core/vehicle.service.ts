import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { VehicleType, Vehicle } from '../shared/vehicle';
import { Paging } from '../shared/paging';

@Injectable({
  providedIn: 'root'
})
export class VehicleService {

  constructor(private httpClient: HttpClient) {
   }

   public searchVehicle(type: VehicleType, pageInfo: Paging): Vehicle[] {
    return null;
   }
}
