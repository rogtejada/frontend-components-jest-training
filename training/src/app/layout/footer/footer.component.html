<footer class="footer bg-dark mt-auto py-3">
    <div class="container">
        <span class="text-muted">All rights reserved to SuperAwesomeVehicleStore. </span>
    </div>
</footer>