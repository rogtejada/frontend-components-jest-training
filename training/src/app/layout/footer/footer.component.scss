.container {
    margin-top: 40px;
    text-align: center;
    border: 0px transparent;    
}
  