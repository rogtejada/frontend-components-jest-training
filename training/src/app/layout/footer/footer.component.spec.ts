import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { FooterComponent } from './footer.component';

describe('FooterComponent', () => {
  let component: FooterComponent;
  let fixture: ComponentFixture<FooterComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ FooterComponent ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(FooterComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  describe('Unit tests', () => {
    it('should create', () => {
      expect(component).toBeTruthy();
    });
  });

  describe('DOM tests', () => {
    it('the footer element should exist', () => {
      const footer = fixture.debugElement.nativeElement.querySelector('footer');
      expect(footer).toBeTruthy();
    });
  });

  describe('Snapshot tests', () => {
    it('Snapshot of html component=', () => {
      const footer = fixture.debugElement.nativeElement.querySelector('footer');
      expect(footer).toMatchSnapshot('FooterComponent');
    });
  });
});
