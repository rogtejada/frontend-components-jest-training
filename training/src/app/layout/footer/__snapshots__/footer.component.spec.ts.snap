// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterComponent Snapshot tests Snapshot of html component=: FooterComponent 1`] = `
<footer
  class="footer bg-dark mt-auto py-3"
>
  <div
    class="container"
  >
    <span
      class="text-muted"
    >
      All rights reserved to SuperAwesomeVehicleStore. 
    </span>
  </div>
</footer>
`;
